core.spec.md

```diff
@@ -0,0 +1,501 @@
+
+###! as: String
+
+Change the [names](#sec-Prefixing) of directives and schema elements from this specification. The specified string MUST be a valid GraphQL identifier and MUST NOT contain the namespace separator (two underscores, {"__"}).
+
+When [`as:`](#@core/as) is provided, processors MUST replace the default name prefix on the names of all [prefixed schema elements](#sec-Elements-which-must-be-prefixed) with the specified name.
+
+```graphql example -- Using {@core}(feature:, as:) to use a feature with a custom name
+schema
+  @core(feature: "https://spec.example.com/core/v1.0")
+  @core(feature: "https://spec.example.com/example/v1.0", as: "eg")
+{
+  query: Query
+}
+
+type User {
+  # Specifying `as: "eg"` transforms @example into @eg
+  name: String @eg(data: ITEM)
+}
+
+# Additional specified schema elements must have their prefixes set
+# to the new name.
+#
+# This data enum was specified as `example__Data`, but will be renamed
+# as `eg__Data`:
+enum eg__Data {
+  ITEM
+}
+
+# Name transformation must also be applied to definitions pulled in from
+# specifications.
+directive @eg(data: eg__Data) on FIELD
+
+# (...other definitions omitted...)
+```
+
+###! export: Boolean! = false
+
+Control whether the entire feature is exported to the API.
+
+If {true}, all schema elements--including all directive usages--from this feature will be included in the schema's API *unless* they are annotated with {@core__export}`(export: false)`.
+
+If {false} (the default), no schema elements from this feature will be included in the schema's API *unless* they are annotated with {@core__export}.
+
+##! @core__export
+
+Specify whether the annotated element is exported to the API.
+
+```graphql definition
+directive @core__export(export: Boolean! = true)
+  repeatable on
+  | SCALAR
+  | OBJECT
+  | FIELD_DEFINITION
+  | ARGUMENT_DEFINITION
+  | INTERFACE
+  | UNION
+  | ENUM
+  | ENUM_VALUE
+  | INPUT_OBJECT
+  | INPUT_FIELD_DEFINITION
+```
+
+{@core__export} can occur at any type system location. Elements with {@core__export} will always be included in the API. Elements with {@core__export}`(export: false)` will always be excluded from the API.
+
+###! export: Boolean
+
+If true, the element is always exported, regardless of whether the feature which defines it [is exported](#@core/export). If false, the element is never exported.
+
+# Prefixing
+
+With the exception of a single root directive, core feature specifications MUST prefix all schema elements they introduce. The prefix:
+  1. MUST match the default name of the feature as derived from the feature's specification URL,
+  2. MUST be a string of characters valid within GraphQL names, and
```

Comment:
SUBSTANTIVE: Just say that it must be a Name (https://spec.graphql.org/draft/#Name). "string of characters valid within" is not precise because the rules for the first character differ from the rules for subsequent characters (ie, is "123" a valid name?)